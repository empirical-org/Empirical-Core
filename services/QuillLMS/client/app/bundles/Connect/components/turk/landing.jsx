import React from 'react'

export default class TurkLanding extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      showIntro: false,
    }
  }

  handleBeginClick = () => {
    const { lesson, begin, } = this.props

    if (lesson.landingPageHtml) {
      this.setState({showIntro: true})
    } else {
      begin()
    }
  }

  handleStartLessonClick = () => {
    const { begin, } = this.props
    begin()
  }

  render() {
    const { showIntro, } = this.state
    const { lesson, } = this.props
    if (showIntro) {
      return (
        <div className="container">
          <div className="landing-page-html" dangerouslySetInnerHTML={{__html: lesson.landingPageHtml}} />
          <button className="quill-button-archived focus-on-light large primary contained" onClick={this.handleStartLessonClick} type="button">Start activity</button>
        </div>
      )
    } else {
      return (
        <div className="landing-page">
          <h1>You&#39;re testing new Quill Activities </h1>
          <p>
            You&#39;re about to answer questions about writing sentences.
            Please answer to the best of your ability.
          </p>
          <button className="quill-button-archived focus-on-light primary contained large" onClick={this.handleBeginClick} type="button">Begin</button>
        </div>
      )
    }
  }

}
