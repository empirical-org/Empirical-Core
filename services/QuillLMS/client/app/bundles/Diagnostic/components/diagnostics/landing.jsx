import React from 'react';

import { ResumeOrBeginButton } from '../../../Shared/index';

export default class Landing extends React.Component {

  resume = () => {
    const { resumeActivity, session, } = this.props
    resumeActivity(session);
  }

  renderButton = () => {
    const { session, begin, } = this.props
    let onClickFn,
      text;
    if (session) {
      // resume session if one is passed
      onClickFn = this.resume;
      text = <span>Resume</span>;
    } else {
      // otherwise begin new session
      onClickFn = begin;
      text = <span>Begin</span>;
    }
    return (
      <ResumeOrBeginButton onClickFn={onClickFn} text={text} />
    );
  }

  getLandingPageHTML = () => {
    const { landingPageHtml, questionCount, } = this.props
    if (landingPageHtml && landingPageHtml !== '<br/>') {
      return landingPageHtml
    } else {
      return (
        <div>
          <h1>You&#39;re working on the Quill Placement Activity </h1>
          <p>
          You&#39;re about to answer {questionCount || '22'} questions about writing sentences.
          Don&#39;t worry, it&#39;s not a test. It&#39;s just to figure out what you know.
          </p>
          <p className="second-p">
          Some of the questions might be about things you haven&#39;t learned yetâ€”that&#39;s okay!
          Just answer them as best as you can.
          Once you&#39;re finished, Quill will create a learning plan just for you!
          </p>
        </div>
      )
    }
  }

  render() {
    return (
      <div className="landing-page">
        <div dangerouslySetInnerHTML={{ __html: this.getLandingPageHTML(), }} />
        {this.renderButton()}
      </div>
    );
  }

}
