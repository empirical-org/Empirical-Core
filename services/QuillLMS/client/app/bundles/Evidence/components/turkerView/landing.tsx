import * as React from "react";
const checkIcon = <img alt="check icon" src={`${process.env.CDN_URL}/images/icons/check-circle-big.svg`} />;

const ListElement = (text: string, textArray?: string[], style?: string) => {
  const textElement = textArray ? <div className={style}>{textArray[0]}<p>{text}</p>{textArray[1]}</div> : <p>{text}</p>;
  return(
    <li>
      {checkIcon}
      {textElement}
    </li>
  );
}

export const TurkLanding = ({ handleStartActivity}) => {
  return(
    <section className="turk-landing-container">
      <section className="turk-landing-header-container">
        <h1>Quill Activity - Beta Testing of New Content</h1>
        <p>You are completing a reading and writing activity. Your responses will be used to develop our system so that millions of students around the world can improve their reading and writing skills.</p>
      </section>
      <section className="turk-landing-directions-container">
        <h1>Directions:</h1>
        <div className="bolded">You should:</div>
        <ul>
          {ListElement('Read the passage')}
          {ListElement('Complete all three sentences')}
          {ListElement('Analyze the feedback provided')}
          {ListElement('five', ['Use the feedback to revise each sentence up to', ' times'])}
        </ul>
        <div className="bolded">Each of your sentences <p>must</p></div>
        <ul>
          {ListElement('one', ['Be ', ' sentence long'])}
          {ListElement('Use proper capitalization (i.e. do not write in capital letters)')}
          {ListElement('Be based on the text, but written in your own words')}
        </ul>
      </section>
      <section className="button-container">
        <button
          className="quill-button small primary contained focus-on-light"
          onClick={handleStartActivity}
          type="button"
        >Start Activity</button>
      </section>
    </section>
  )
}

export default TurkLanding;
