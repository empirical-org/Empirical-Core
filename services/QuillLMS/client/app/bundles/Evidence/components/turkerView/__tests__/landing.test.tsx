import * as React from 'react';
import { shallow } from 'enzyme';

import { TurkLanding } from '../landing';

describe('TurkLanding Component', () => {
  const component = shallow(<TurkLanding handleStartActivity={jest.fn} />);

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
