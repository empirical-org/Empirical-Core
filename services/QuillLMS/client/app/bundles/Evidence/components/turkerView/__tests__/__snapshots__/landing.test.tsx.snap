// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TurkLanding Component should match snapshot 1`] = `
<section
  className="turk-landing-container"
>
  <section
    className="turk-landing-header-container"
  >
    <h1>
      Quill Activity - Beta Testing of New Content
    </h1>
    <p>
      You are completing a reading and writing activity. Your responses will be used to develop our system so that millions of students around the world can improve their reading and writing skills.
    </p>
  </section>
  <section
    className="turk-landing-directions-container"
  >
    <h1>
      Directions:
    </h1>
    <div
      className="bolded"
    >
      You should:
    </div>
    <ul>
      <li>
        <img
          alt="check icon"
          src="undefined/images/icons/check-circle-big.svg"
        />
        <p>
          Read the passage
        </p>
      </li>
      <li>
        <img
          alt="check icon"
          src="undefined/images/icons/check-circle-big.svg"
        />
        <p>
          Complete all three sentences
        </p>
      </li>
      <li>
        <img
          alt="check icon"
          src="undefined/images/icons/check-circle-big.svg"
        />
        <p>
          Analyze the feedback provided
        </p>
      </li>
      <li>
        <img
          alt="check icon"
          src="undefined/images/icons/check-circle-big.svg"
        />
        <div>
          Use the feedback to revise each sentence up to
          <p>
            five
          </p>
           times
        </div>
      </li>
    </ul>
    <div
      className="bolded"
    >
      Each of your sentences 
      <p>
        must
      </p>
    </div>
    <ul>
      <li>
        <img
          alt="check icon"
          src="undefined/images/icons/check-circle-big.svg"
        />
        <div>
          Be 
          <p>
            one
          </p>
           sentence long
        </div>
      </li>
      <li>
        <img
          alt="check icon"
          src="undefined/images/icons/check-circle-big.svg"
        />
        <p>
          Use proper capitalization (i.e. do not write in capital letters)
        </p>
      </li>
      <li>
        <img
          alt="check icon"
          src="undefined/images/icons/check-circle-big.svg"
        />
        <p>
          Be based on the text, but written in your own words
        </p>
      </li>
    </ul>
  </section>
  <section
    className="button-container"
  >
    <button
      className="quill-button small primary contained focus-on-light"
      onClick={[Function]}
      type="button"
    >
      Start Activity
    </button>
  </section>
</section>
`;
